### PR TITLE
Adding cluster size to track extra

### PR DESCRIPTION
**Pull request by fmazzasc**
Follows the discussion presented in https://indico.cern.ch/event/1332124/
Cluster sizes are added to AO2Ds and cluster map is removed. Few modifications have been done to the ITS tracking and ITSTPC matching to get the cl size information (pinging @mconcas , @mpuccio , @shahor02 ). We have also updated the AnalysisDataModel and created a versioning of the trackExtra tables. Few dyn columns have been changed, I will undraft the PR as soon as I checked that we get consistent results with the old dataformat (pinging @ddobrigk , @jgrosseo )

### CODE REVIEW

**Review by shahor02 — DISMISSED** [review]
cannot judge about the backward compatibility of the AOD track model changes, the rest looks ok.

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h:255]: As DetectorMap is part of the old and the new version, we also need an old and a new version of this column

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
DECLARE_SOA_TABLE_FULL(StoredTracksExtra_000, "TracksExtra", "AOD", "TRACKEXTRA", //! On disk version of TracksExtra, version 0
```

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
DECLARE_SOA_TABLE_FULL_VERSIONED(StoredTracksExtra_001, "TracksExtra", "AOD", "TRACKEXTRA", 1, // On disk version of TracksExtra, version 1
```

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: Does this work with the same name "TRACKEXTRA" ?

Comment (jgrosseo) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: I guess the second needs to have _001 but I may be wrong

**Review by jgrosseo — CHANGES_REQUESTED** [review]
Thanks for this work!

For now, I have only reviewed the changes in AnalysisDataModel. See my minor comments.

However, we have to split this PR in two PRs. The first one introduces the changes in the AnalysisDataModel **only**, but keeps version 0 as default.
Then when it is merged, you make the PR with the converter in O2Physics
Once that is merged, you make a PR which contains the rest and the change to version 1.

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (aalkin) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: No, the `description` field should stay the same. The version is added to the tree name if the table is written out.

File: Framework/Core/src/AODReaderHelpers.cxx
Comment (aalkin) [inline Framework/Core/src/AODReaderHelpers.cxx]: This does not seem correct, the version is encoded in subspec, not in description. I see that this is also the case for the mc particles table, which is also incorrect, it just works by accident because the extensions are identical. I will fix the case with mc particles, but here you should do the following
```cpp
if (description == header::DataDescription{"TRACKEXTRA"}) {
    if (version == 0U) {
    } else if (version == 1U) {
    }
}
```

**Comment by fmazzasc** [comment]
Thanks to all of you for the comments and the help.
- @jgrosseo : I implemented all your comments and switched to 000 as default. As soon as the converter in O2Physics is merged I will move to  001 
- @aalkin : I changed the data decription as you suggested

I also checked the itsClusterMap with 000 and 001, and the dynamic evaluation works smoothly (see picture). Opening the PR.
Screenshot from 2023-10-11 12-37-57

**Comment by mconcas** [comment]
ITS side small changes also look ok.

**Comment by jgrosseo** [comment]
Don't you want to factor our the AnalysisDataModel changes in a separate PR and then merge this one once you swap to version 1 and can activate the filling in the AOD producer instead of the commenting out code?

**Comment by fmazzasc** [comment]
> Don't you want to factor our the AnalysisDataModel changes in a separate PR and then merge this one once you swap to version 1 and can activate the filling in the AOD producer instead of the commenting out code?

I removed the commented code and opened in draft mode https://github.com/AliceO2Group/AliceO2/pull/12087

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]: I am sorry that we seem not to converge. ;-)

I think now where we have the `v001` namespace, we should move this there and call it `ITSNCls` and not `ITSNCls001`. The same for `ITSClusterMapDyn`. What do you think?

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
DECLARE_SOA_COLUMN(ITSClusterMap, itsClusterMap, uint8_t);                                    //! Old cluster ITS cluster map, kept for version 0 compatibility
```

File: Framework/Core/test/test_AnalysisTask.cxx
Comment (jgrosseo) [inline Framework/Core/test/test_AnalysisTask.cxx:167]: This has to be changed again when we move to version 1. Is this the best way to implement the test?

Comment (fmazzasc) [reply Framework/Core/test/test_AnalysisTask.cxx:167]: Simple `TracksExtraExtension` does not work neither with 000 nor 001. Also for Collisions the test follows the same syntax

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (fmazzasc) [reply Framework/Core/include/Framework/AnalysisDataModel.h]: I agree. Done!

File: Framework/Core/include/Framework/AnalysisDataModel.h
Comment (jgrosseo) [inline Framework/Core/include/Framework/AnalysisDataModel.h]:
```suggestion
DECLARE_SOA_EXPRESSION_COLUMN(DetectorMap, detectorMap, uint8_t, //! Detector map version 1, see enum DetectorMapEnum
```

**Review by jgrosseo — DISMISSED** [review]

**Comment by fmazzasc** [comment]
@shahor02 , @aalkin , could we merge this PR?

File: Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h
Comment (shahor02) [inline Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h]: why this needs to be int?

File: Detectors/GlobalTracking/include/GlobalTracking/MatchTPCITS.h
Comment (shahor02) [inline Detectors/GlobalTracking/include/GlobalTracking/MatchTPCITS.h]: why this needs to be int?

File: DataFormats/Detectors/ITSMFT/ITS/include/DataFormatsITS/TrackITS.h
Comment (shahor02) [inline DataFormats/Detectors/ITSMFT/ITS/include/DataFormatsITS/TrackITS.h]: strange that this passed the check, could you put brackets?

File: Framework/Core/test/test_AnalysisTask.cxx
Comment (aalkin) [reply Framework/Core/test/test_AnalysisTask.cxx:167]: Binding can be indeed extracted from the type, I will update the test outside this PR.

File: DataFormats/Detectors/ITSMFT/ITS/include/DataFormatsITS/TrackITS.h
Comment (fmazzasc) [reply DataFormats/Detectors/ITSMFT/ITS/include/DataFormatsITS/TrackITS.h]: Added

File: Detectors/GlobalTracking/include/GlobalTracking/MatchTPCITS.h
Comment (fmazzasc) [reply Detectors/GlobalTracking/include/GlobalTracking/MatchTPCITS.h]: Moved to unsigned short

File: Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h
Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h]: moved to unsigned short

Comment (shahor02) [reply Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h]: you anyway round it to max 15, right? Could go even to `int8_t`

Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h]: Done. I needed to add an `if` for the treatement of overflows

Comment (fmazzasc) [reply Detectors/ITSMFT/ITS/tracking/include/ITStracking/TimeFrame.h]: Can we merge it now?

**Review by shahor02 — DISMISSED** [review]

File: DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/TrkClusRef.h
Comment (ktf) [inline DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/TrkClusRef.h:28]: Shouldn't we swap the order of `pattern` and `clsizes` to avoid wasting 16 bits in padding per ref?

Comment (fmazzasc) [reply DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/TrkClusRef.h:28]: done, thanks for spotting!

**Comment by shahor02** [comment]
checks were passed before the trivial swap, merging

**Pull request merged**
